### PR TITLE
Create patient edit/form

### PR DESCRIPTION
**Pull request by issa-diallo**

### CODE REVIEW

File: src/components/PatientForm.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+}
+
+interface Form {
+  lastname: Field
+  firstname: Field
+  address: Field
+  city: Field
+  zipcode: Field
+
+}
```
Comment (AndreMiras) [inline src/components/PatientForm.tsx:30]: Can't we simply use the Patient already defined in types?
The only thing I can think of that could be an issue is it's expecting an ID and I don't know how that would play when using the `updatePatient()` function

File: src/components/PatientForm.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    patient.city = form.city.value
+    patient.zip_code = form.zipcode.value
+    await onUpdate()
+    history(`/patients/${patient.id}`)
+  }
+
+  return (
+    <form onSubmit={async e => await handleSubmit(e)}>
+      <Row>
+        <div className='col s-12 m-8 offset-m-2'>
```
Comment (AndreMiras) [inline src/components/PatientForm.tsx:82]: Do we really need these two?

File: src/components/PatientForm.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+          <div className='card hoverable'>
```
Comment (AndreMiras) [inline src/components/PatientForm.tsx:83]: Maybe we don't actually need a card here as it kind of overload the final rendering.
I mean we have extra borders that overlaps with the form control ones.
Maybe we should follow the same layout as the profile page to keep it consistent

File: src/components/PatientForm.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+import Row from 'react-bootstrap/Row'
+import { updatePatient } from '../services/api'
+
+interface PatientFormProps {
+  patient: Patient
+}
+
+interface Field {
+  value: any
+  error?: string
```
[the remark this answers is not in the document]
Comment (issa-diallo) [reply src/components/PatientForm.tsx:19]: fix

File: src/components/PatientForm.tsx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    await onUpdate()
+    history(`/patients/${patient.id}`)
+  }
+
+  return (
+    <form onSubmit={async e => await handleSubmit(e)}>
+      <Row>
+        <div className='col s-12 m-8 offset-m-2'>
+          <div className='card hoverable'>
+            <div className='card-content'>
```
[the remark this answers is not in the document]
Comment (issa-diallo) [reply src/components/PatientForm.tsx:84]: fix

File: src/components/PatientForm.tsx [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+              {/* patient lastname */}
+              <div className='form-group'>
+                <label htmlFor='lastname'>Nom</label>
+                <input id='lastname' name='lastname' type='text' className='form-control' value={form.lastname.value} onChange={(e) => handleChange(e)} />
+              </div>
```
[the remark this answers is not in the document]
Comment (issa-diallo) [reply src/components/PatientForm.tsx:89]: fix

File: src/components/PatientForm.tsx [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+  }
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>): Promise<void> => {
+    e.preventDefault()
+    patient.firstname = patientState.firstname
+    patient.lastname = patientState.lastname
+    patient.address = patientState.address
+    patient.city = patientState.city
+    patient.zip_code = patientState.zip_code
+    patient.phone = patientState.phone
```
Comment (AndreMiras) [inline src/components/PatientForm.tsx:56]: not needed, we can consume the patientState directly in the `onUpdate()`

File: src/pages/PatientEdit.tsx [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+  useEffect(() => {
+    if (token === null) return
+    // eslint-disable-next-line no-void
+    void (async () => await fetchPatientCallback())()
+  }, [token, fetchPatientCallback])
+
+  return (
+    <>
+      {(patient != null) ? (
```
Comment (AndreMiras) [inline src/pages/PatientEdit.tsx:44]: `!==`

**Pull request merged**